File: Java/WordFrequencyCounter.java
package Java;

import java.util.HashMap;
import java.util.Map;

public class WordFrequencyCounter {
  private static void countWordFrequency(String text) {
    Map<String, Integer> wordFrequency = new HashMap<>();
    String[] textArray = text.split(" ");

    for (String word : textArray) {
      if (!wordFrequency.containsKey(word)) {
        wordFrequency.put(word, 1);
      } else {
        wordFrequency.put(word, wordFrequency.get(word) + 1);
      }
    }

    System.out.println(wordFrequency.values());
  }

  public static void main(String[] args) {
    countWordFrequency("banana aipim banana macaco banana maça");
    countWordFrequency("banana macaco banana aipim banana macaco");
  }
}

File: TS/wordFrequencyCounter.ts
function wordFrequencyCounter(text: string): void {
  const array = text.split(" ");
  let map = new Map<string, number>();

  for (let word of array) {
    if (!map.has(word)) {
      map.set(word, 1);
    } else {
      map.set(word, map.get(word)! + 1);
    }
  }

  console.log(map.entries());
  
}

wordFrequencyCounter("banana aipim banana macaco banana maça");
wordFrequencyCounter("banana macaco banana aipim banana mamaco");